src/test/scala/mesosphere/marathon/core/readiness/ReadinessCheckSpecTest.scala

```diff
@@ -0,0 +1,142 @@
+
+import mesosphere.marathon.MarathonTestHelper
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.{ PortDefinition, PathId, AppDefinition }
+import org.scalatest.{ GivenWhenThen, Matchers, FunSuite }
+import scala.collection.immutable.Seq
+
+class ReadinessCheckSpecTest extends FunSuite with Matchers with GivenWhenThen {
+  import MarathonTestHelper.Implicits._
+
+  test("readiness check specs for one task with dynamic ports and one readiness check") {
+    val f = new Fixture
+
+    Given("an app with a readiness check and a randomly assigned port")
+    val app = f.appWithOneReadinessCheck
+    And("a task with two host port")
+    val task = f.taskWithPorts
+
+    When("calculating the ReadinessCheckSpec")
+    val specs = ReadinessCheckExecutor.ReadinessCheckSpec.readinessCheckSpecsForTask(app, task, task.launched.get)
+
+    Then("we get one spec")
+    specs should have size 1
+    val spec = specs.head
+    And("it has the correct url")
+    spec.url should equal("http://host.some:80/")
+    And("the rest of the fields are correct, too")
+    spec should equal(
+      ReadinessCheckExecutor.ReadinessCheckSpec(
+        url = "http://host.some:80/",
+        taskId = task.taskId,
+        checkName = app.readinessChecks.head.name,
+        interval = app.readinessChecks.head.interval,
+        timeout = app.readinessChecks.head.timeout,
+        httpStatusCodesForReady = app.readinessChecks.head.httpStatusCodesForReady,
+        preserveLastResponse = app.readinessChecks.head.preserveLastResponse
+      )
+    )
+  }
+
+  test("readiness check specs for one task with fixed ports and one readiness check") {
+    val f = new Fixture
+
+    Given("an app with a readiness check and a fixed port assignment")
+    val app = f.appWithOneReadinessCheckWithFixedPort
+    And("a task with two host port")
+    val task = f.taskWithPorts
+
+    When("calculating the ReadinessCheckSpec")
+    val specs = ReadinessCheckExecutor.ReadinessCheckSpec.readinessCheckSpecsForTask(app, task, task.launched.get)
+
+    Then("we get one spec")
+    specs should have size 1
+    val spec = specs.head
+    And("it has the correct url")
+    spec.url should equal("http://host.some:8080/")
+    And("the rest of the fields are correct, too")
+    spec should equal(
+      ReadinessCheckExecutor.ReadinessCheckSpec(
+        url = "http://host.some:8080/",
+        taskId = task.taskId,
+        checkName = app.readinessChecks.head.name,
+        interval = app.readinessChecks.head.interval,
+        timeout = app.readinessChecks.head.timeout,
+        httpStatusCodesForReady = app.readinessChecks.head.httpStatusCodesForReady,
+        preserveLastResponse = app.readinessChecks.head.preserveLastResponse
+      )
+    )
+  }
+
+  test("multiple readiness check specs") {
+    val f = new Fixture
+
+    Given("an app with two readiness checks and randomly assigned ports")
+    val app = f.appWithMultipleReadinessChecks
```

Comment:
at least in this code, I chose to do that. I don't mind if other code doesn't and we prevent this by validation, though.